src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticStartAnalysisScope.cs

```diff
@@ -608,6 +608,67 @@ public void RegisterSymbolAction(DiagnosticAnalyzer analyzer, Action<SymbolAnaly
             SymbolAnalyzerAction analyzerAction = new SymbolAnalyzerAction(action, symbolKinds, analyzer);
             this.GetOrCreateAnalyzerActions(analyzer).AddSymbolAction(analyzerAction);
             _symbolActions = _symbolActions.Add(analyzerAction);
+
+            // The SymbolAnalyzerAction does not handle SymbolKind.Local or SymbolKind.Parameter
+            // because the compiler does not make CompilationEvents for them. As a workaround,
+            // handle them specially by registering further OperationActions (for 
+            // VariableDeclarations) or SymbolActions (for Methods) and utilize the results of
+            // those to construct the necessary SymbolAnalysisContexts.
+
+            if (symbolKinds.Contains(SymbolKind.Local))
+            {
+                RegisterOperationAction(analyzer, opContext =>
+                {
+                    var varDecl = (IVariableDeclaration)opContext.Operation;
+                    action(new SymbolAnalysisContext(
+                        varDecl.Variable,
+                        opContext.Compilation,
+                        opContext.Options,
+                        opContext.ReportDiagnostic,
+                        opContext.IsSupportedDiagnostic,
+                        opContext.CancellationToken));
+                },
+                ImmutableArray.Create(OperationKind.VariableDeclaration));
+            }
+
+            if (symbolKinds.Contains(SymbolKind.Parameter))
+            {
+                RegisterSymbolAction(
+                    analyzer, 
+                    context => 
+                    {
+                        ImmutableArray<IParameterSymbol> parameters;
+
+                        switch (context.Symbol.Kind)
+                        {
+                            case SymbolKind.Method:
+                                parameters = ((IMethodSymbol)context.Symbol).Parameters;
+                                break;
+                            case SymbolKind.Property:
+                                parameters = ((IPropertySymbol)context.Symbol).Parameters;
+                                break;
+                            case SymbolKind.NamedType:
+                                var namedType = (INamedTypeSymbol)context.Symbol;
+                                var delegateInvokeMethod = namedType.DelegateInvokeMethod;
+                                parameters = delegateInvokeMethod?.Parameters ?? ImmutableArray.Create<IParameterSymbol>();
+                                break;
+                            default:
+                                throw new ArgumentException(nameof(context));
+                        }
+
+                        foreach (var parameter in parameters.Where(p => !p.IsImplicitlyDeclared))
```

Comment:
Note: it would not be clean, but symbols do expose .Language.  That means you can handle the VB implicit parameter issue here by doing a language check.  As this API is purely for C#/VB, i think it's fine for you to do that.